presto-main/src/main/java/com/facebook/presto/tracing/TracingConfig.java

```diff
@@ -23,7 +26,15 @@
         public static final String SIMPLE = "simple";
     }
 
+    public static class DistributedTracingMode
+    {
+        public static final String NO_TRACE = "no_tracing";
+        public static final String ADHOC = "adhoc";
+        public static final String SAMPLE_BASED = "sample_based";
+    }
```

Comment:
This doesn't need t be a class of strings. It could be an enum class. How about "disabled", "enabled", and "sample_based"?

Reply:
Yeah I was trying to be consistent with the TracerType. Is there any issues here for it to be string defined?
"disabled" "enabled" and "sample_based" sound like a good plan

Reply:
Actually I don't think the 3 terms mentioned would be proper after re-thinking. If we are talking about modes, we don't call "enabled" being a mode, right? What do you think?

Reply:
The `TracerType` is a different story. Because you need to feed string values beyond what can be defined by an enum. In this case, we know what the values are. So enum is preferred. Regarding "adhoc", I think that is the most confusing part. Because an adhoc mode means users need to do something to provide an adhoc solution or what adhoc things users should do? So the "adhoc" word itself could be misleading.

Reply:
Cool, enum makes sense. Regarding the tracing mode though can we call them "always_trace", "sample_based", and "no_trace" ?